docs/servicedns-with-externaldns.md

```diff
@@ -0,0 +1,221 @@
+# Federated Service DNS with ExternalDNS Guide
+
+Federated Service DNS (FSDNS) provides the ability to programmatically manage DNS records of Service objects.
+FSDNS is not meant to replace in-cluster DNS providers such as [CoreDNS](https://coredns.io/). Instead, FSDNS works with
+systems such as [ExternalDNS](https://github.com/kubernetes-incubator/external-dns) that are responsible for
+managing external DNS records of Service objects.
+
+## Concepts
+
+<p align="center"><img src="./images/servicedns-with-externaldns.png" width="711"></p>
+
+The above diagram illustrates FSDNS. A typical FSDNS workflow consists of:
+
+1. Creating `FederatedDeployment` and `FederatedService` objects. The Federation sync
+controller propagates the corresponding objects to target clusters.
+2. Creating a `MultiClusterServiceDNSRecord` object that identifies the intended domain name and other optional resource
+record details of the `FederatedService`.
+3. A `DNSEndpoint` object that corresponds to the `MultiClusterServiceDNSRecord` is created by the DNS Endpoint
+Controller.
+3. An external DNS system (i.e. ExternalDNS) watches and lists `DNSEndpoint` objects and creates DNS resource records
+in supported DNS providers based on the desired state of the object.
+
+FSDNS is comprised of multiple types and controllers:
+- The Service DNS Controller will watch and list
+`MultiClusterServiceDNSRecord` objects and update the status of the object with the corresponding Service load-balancer
+IP address for each cluster listed in the `FederatedServicePlacement`.
+- The DNS Endpoint Controller watches and lists the
+`MultiClusterServiceDNSRecord` object and creates a `DNSEndpoint` object for each `MultiClusterServiceDNSRecord`. An
+external DNS system (i.e. [ExternalDNS](https://github.com/kubernetes-incubator/external-dns)) is responsible for
+watching `DNSEndpoint` objects and using the endpoint information to create the DNS resource record in
+the DNS provider.
+
+## User Guide
+
+Follow the steps in the Federation-v2 [User Guide](./userguide.md) to setup your Kubernetes clusters and federation
+control-plane.
+
+ExternalDNS supports several different
+[providers](https://github.com/kubernetes-incubator/external-dns#the-latest-release-v05). If needed, create a domain
+name with one of the supported providers or delegate a DNS subdomain to use for ExternalDNS. Reference your DNS
+provider documentation on how to create a domain or delegate a subdomain. The `example` subdomain of domain name
+`example.com` is used by this guide.
+
+Deploy the ExternalDNS controller. __NOTE:__ [Google Cloud DNS](https://cloud.google.com/dns) is used as the provider
+and `external.example.com` is used as the `domain-filter`. The filter will make ExternalDNS see only the hosted zones
+matching `external.example.com`. Omit this flag to process all available hosted zones.
+```bash
+cat <<EOF | kubectl create -f -
```

Comment:
How about adding a $ to clarify that this is a shell command? Ditto for others.

Reply:
ack